entity-framework/core/managing-schemas/migrations/applying.md

```diff
@@ -320,6 +320,9 @@ It's possible for the application itself to apply migrations programmatically, t
 * It's important to be able to roll back an applied migration in case of an issue. The other strategies provide this easily and out of the box.
 * The SQL commands are applied directly by the program, without giving the developer a chance to inspect or modify them. This can be dangerous in a production environment.
 
+> [!NOTE]
+> EF9 introduced migration locking, which protects against multiple migrations happening simultaneously. However, other problems listed above are relevant and as such applying migration at runtime is still strongly discouraged in production.
```

Comment:
I'd maybe make the wording a bit softer (e.g. remove `strongly`).. I think that once locking is implemented, doing this is maybe not that bad (we can discuss what our policy/recommendation is).

In any case, I'd add "On version of EF prior to 9, " at the start of the first bullet point above.